.github/workflows/pgmq.yml

```diff
@@ -0,0 +1,73 @@
+name: pgmq CI workflow
+
+defaults:
+  run:
+    shell: bash
+    working-directory: ./crates/pgmq/
+
+on:
+  pull_request:
+    branches:
+     - main
+    paths:
+      - '.github/workflows/pgmq.yml'
+      - 'crates/pgmq/**'
+  push:
+    branches:
+      - main
+      - release**
+
+
+jobs:
+  lint:
+    name: Run linters
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install minimal nightly with clippy and rustfmt
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rustfmt, clippy
+      - uses: Swatinem/rust-cache@v2
+      - name: Cargo format
+        run: cargo +nightly fmt --all --check
+      - name: Clippy
+        run: cargo clippy
+
+  tests:
+    name: Run tests
+    runs-on: ubuntu-22.04
+    needs: lint
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - uses: Swatinem/rust-cache@v2
+      - name: unit-tests
+        run: cargo test --lib
+      - name: integration-tests
+        run: make test.integration
+      - name: teardown
+        run: make test.cleanup
+
+  publish:
+    # only publish off tag events
+    if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Hmm this isn't going to work. We need a way to trigger this job to publish release ONLY for this particular crate in the repo. Not all tag/release events will be for this crate.